### PR TITLE
Add some initial documentation

### PR DESCRIPTION
**Pull request by romaninsh**
Some initial documentation for #308. Early feedback is welcome!!

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+Supported Types
+===============
+
+ATK Data prior to 1.5 supports the following types:
+
+ - string
+ - boolean ([':php:class:`Boolean`'])
+ - integer ([':php:class:`Number`', 'precision'=>0])
+ - money ([':php:class:`Number`', 'prefix'=>'€', 'precision'=>2])
+ - float ([':php:class:`Number`', 'type'=>'float'])
```
Comment (DarkSide666) [inline docs/types.rst]: argument name `type` is a bit strange here. type = float
maybe use precision = 'float' instead for Number class?

File: docs/types.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+ - date ([':php:class:`DateTime`'])
+ - datetime ([':php:class:`DateTime`'])
+ - time ([':php:class:`DateTime`'])
+ - password ([':php:class:`Password`])
```
Comment (DarkSide666) [inline docs/types.rst]: maybe can pass password hashing method here in arguments?

File: docs/types.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+ - percent (34.2%) ([':php:class:`Number`', 'format'=>function($v){ return $v*100; }, 'postfix'=>'%'])
+ - rating (3 out of 5) ([':php:class:`Number`', 'max'=>5, 'precision'=>0])
+ - uuid (xxxxxxxx-xxxx-...) ([':php:class:`Number`', 'base'=>16, 'mask'=>'########-##..'])
+ - hex (number with base 16) ([':php:class:`Number`', 'base'=>16])
+ - ip (123.2.44.1) ([':php:class:`Number`', 'base'=>256, 'mask'=>'#.#.#.#'])
+ - ipv6 ([':php:class:`Number`', 'base'=>16', 'mask'=>'####:####:..']);
+ - model (used for containment)
+ - fraction (5/7) ([':php:class:`Fraction`'])
+
```
Comment (DarkSide666) [inline docs/types.rst]: + location (latitude, longitude) - class Location

File: docs/types.rst (hunk 1 above)
Comment (romaninsh) [reply docs/types.rst]: i think it's mainly for backward compatibility. Certainly you can specify number and precision also.

File: docs/types.rst (hunk 2 above)
Comment (romaninsh) [reply docs/types.rst]: password hashing is part of https://github.com/atk4/login.

File: README.md [hunk 4]
```diff
@@ -1,18 +1,27 @@
-# Agile Data
+# ATK Data - Data Access framework for High-Latency databases
+
+The reason you hate Object Relational Mapper (ORM) is because it's slow, clumsy, limiting and flawed. The reason you *have to* use it is because of consistency, compatibility and abstraction.
+
+**ATK Data provides a PHP framework, an alternative to ORM, which comes with same and more benefits without inheriting any of the design flaws.**
+
+ATK Data is focused on reducing number of database queries, moving CPU-intensive tasks into your database (if possible). It is well suited for Amazon RDS, Google Cloud SQL and ClearDB but thanks to abstraction will work transparently with Static data, NoSQL or RestAPIs backends.
```
Comment (DarkSide666) [inline README.md]: No MySQL, PostGRE and Oracle mentioned ? :)

File: README.md [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
 [![Build Status](https://travis-ci.org/atk4/data.png?branch=develop)](https://travis-ci.org/atk4/data)
 [![Code Climate](https://codeclimate.com/github/atk4/data/badges/gpa.svg)](https://codeclimate.com/github/atk4/data)
 [![Test Coverage](https://codeclimate.com/github/atk4/data/badges/coverage.svg)](https://codeclimate.com/github/atk4/data/coverage)
 [![codecov](https://codecov.io/gh/atk4/data/branch/develop/graph/badge.svg)](https://codecov.io/gh/atk4/data)
 
+[![GitHub release](https://img.shields.io/github/release/atk4/data.svg)](CHANGELOG.md)
 
-**Data Access Framework for high-latency databases (Cloud SQL/NoSQL).**
+Quick-Links: [Documentation](http://agile-data.readthedocs.io). [Namespaces](http://www.agiletoolkit.org/dox/namespaces.html). [Example](https://github.com/atk4/data-primer). [ATK Data](https://github.com/atk4/ui). [Forum](https://forum.agiletoolkit.org/). [Chat](https://gitter.im/atk4/atk4). [Commercial support](https://www.agiletoolkit.org/contact). [Udemy Course](https://forum.agiletoolkit.org/t/udemy-com-atk-course-early-access-limited-time-free/413).
```
Comment (DarkSide666) [inline README.md]: Should be ATK UI instead of ATK Data

File: README.md [hunk 6]
```diff
@@ -24,17 +33,11 @@ Designed for medium to large PHP applications and frameworks, Agile Data is a cl
...
-Alternatively you can watch hands-on screencasts on our Youtube.
-
-**WARNING** - Agile Data has a different architecture to other PHP ORMs, so even if you are fluent with Doctrine or Yii ORM, you should review the core principles.
+### Getting Started
 
-## [Full Documentation](http://agile-data.readthedocs.io/)
+Watch [Quick Start](http://agile-data.readthedocs.io/en/develop/quickstart.html) or [Screencasts](https://www.youtube.com/watch?v=o16xwkFfnuA&t=182s&index=1&list=PLUUKFD-IBZWaaN_CnQuSP0iwWeHJxPXKS). There is also our [Official Udemy Course](https://forum.agiletoolkit.org/t/udemy-com-atk-course-early-access-limited-time-free/413) and [Full Documentation](http://agile-data.readthedocs.io) ([PDF](https://media.readthedocs.org/pdf/agile-data/develop/agile-data.pdf)). 
 
-If you like you can also grab 100+ pages of [documentation in PDF format](https://media.readthedocs.org/pdf/agile-data/develop/agile-data.pdf). This does not include the UI or Core documentations which you can find here:
+ATK Data relies on ATK Core and ATK UI:
```
Comment (DarkSide666) [inline README.md]: ATK Data doesn't rely on ATK UI !

**Pull request merged**